darwin/DarwinProcess.c

```diff
@@ -37,7 +37,7 @@ DarwinProcess* DarwinProcess_new(Settings* settings) {
    this->stime = 0;
    this->taskAccess = true;
 
-   return this;
+   return &this->super;
```

Comment:
Use some central `AsProcess` magic that enforces `&(*this)` and `&this->super` to be the same pointers (mod type).

Reply:
All the Object pseudo inheritance bases on the fact that a struct address equals the address of the first field.

Reply:
It's guaranteed by the C Standard: 

6.7.2.1-13 states:
> Apointer to astructure  object,  suitably  converted,  points  to  its  initial  member  (or  if  that  member  is  abit-field,  then  to  the  unit  in  which  it  resides),  and  vice  versa.   Theremay  be  unnamedpadding within a structure object, but not at its beginning.

http://www.open-std.org/jtc1/sc22/wg14/www/docs/n1256.pdf page 115